src/realizations/catchment/Bmi_Multi_Formulation.cpp

```diff
@@ -205,13 +269,69 @@ const string &Bmi_Multi_Formulation::get_config_mapped_variable_name(const strin
     return output_var_name;
 }
 
+// TODO: remove from this level - it belongs (perhaps) as part of the ForcingProvider interface, but is general to it
 const string &Bmi_Multi_Formulation::get_forcing_file_path() const {
-    // TODO: add something that ensures these are set to same path for all modules (allowing some to be unset also)
     return modules[0]->get_forcing_file_path();
 }
 
-const vector<std::string> &Bmi_Multi_Formulation::get_output_variable_names() const {
-    return modules.back()->get_output_variable_names();
+string Bmi_Multi_Formulation::get_output_line_for_timestep(int timestep, std::string delimiter) {
```

Comment:
This is a bit of a complicated mess.  Definitely worth coming back to and refactoring this behavior.

Reply:
That is why the original `Forcing` concept existed all the way up the chain...I think we don't want as much of a unified interface for data access for any eventual representation that may do data transformations (models...).

Reply:
No, I cannot.  My hasty fingers lied to the keyboard.  Edited the previous comment to **maybe** be more clear...